File: C# Examples/17 PDM Professional Files and Variables/PDMVariables/SolidWorksMacro.cs
using System;
using System.Windows.Forms;
using EPDM.Interop.epdm;
using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace PDMVariables
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            // Create a file vault interface and log in to a vault
            IEdmVault5 vault = new EdmVault5();
            vault.LoginAuto("PDMVault", 0);            

            //Let the user select one or more files that are
            //in the file vault to which we are logged in  
            EdmStrLst5 PathList;
            PathList = vault.BrowseForFile(0, bsCaption: "Select a file to show its information");

            //Check if the user pressed Cancel
            if (PathList == null)
            {
                //do nothing - user pressed cancel
            }
            else
            {
                string message;
                IEdmPos5 pos;
                //Display a message box with 
                //the paths of all selected files.   
                message = "You selected the following files:\n";
                pos = PathList.GetHeadPosition();
                while (!pos.IsNull)
                {
                    string filePath;
                    filePath = PathList.GetNext(pos);
                    //connect to the file object
                    IEdmFile5 eFile;
                    IEdmFolder5 eFolder;
                    eFile = vault.GetFileFromPath(filePath, out eFolder);
                    message += filePath;
                    message += "\nState: " + eFile.CurrentState.Name;
                    message += "\nIs Checked Out: " + eFile.IsLocked.ToString();

                    //get the file's EnumeratorVariable interface
                    //for working with its card variables
                    IEdmEnumeratorVariable10 eVar;
                    eVar = (IEdmEnumeratorVariable10)eFile.GetEnumeratorVariable();

                    //get the description
                    object varValue;
                    eVar.GetVar("Description", "@", out varValue);
                    message += "\nDescription: " + varValue.ToString();

                    //try to check out the file
                    try
                    {
                        eFile.LockFile(eFolder.ID, 0);
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine(ex.Message);
                        Console.WriteLine(eFile.Name);
                        return;
                    }

                    //edit the file here
                    //reconnec to the file's EnumeratorVariable after check out
                    eVar = (IEdmEnumeratorVariable10)eFile.GetEnumeratorVariable();
                    try
                    {
                        eVar.SetVar("Description", "@", "NEW DESCRIPTION");
                        //close the file and save (flush) any updates
                        eVar.CloseFile(true);
                    }
                    catch (Exception ex)
                    {
                        MessageBox.Show(ex.Message + "\n" + eFile.Name);
                        return;
                    }

                    try
                    {
                        eFile.UnlockFile(0, "API check in");
                    }
                    catch (Exception ex)
                    {
                        MessageBox.Show(ex.Message + "\n" + eFile.Name);
                        return;
                    }

                    //show information about the file
                    MessageBox.Show(message);
                }
            }
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 17 PDM Professional Files and Variables/PDMVariables/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System
Imports EPDM.Interop.epdm

Partial Class SolidWorksMacro
    Public Sub main()
        'Create a file vault interface and log in to a vault
        Dim vault As EdmVault5 = New EdmVault5
        Dim vaultName As String = "PDMVault"
        vault.LoginAuto(vaultName, 0)

        'Let the user select one or more files that are
        'in the file vault to which we are logged in  
        Dim PathList As EdmStrLst5
        PathList = vault.BrowseForFile(0, , , , , ,
    "Select a file to show its information")

        'Check if the user pressed Cancel
        If PathList Is Nothing Then
            'do nothing - user pressed cancel
        Else
            Dim message As String
            Dim pos As IEdmPos5
            'Display a message box with 
            'the paths of all selected files.        
            message = "You selected the following files:" + vbLf
            pos = PathList.GetHeadPosition
            While Not pos.IsNull
                Dim filePath As String
                filePath = PathList.GetNext(pos)

                'connect to the file object
                Dim eFile As IEdmFile5
                Dim parentFolder As IEdmFolder5
                eFile = vault.GetFileFromPath(filePath, parentFolder)
                message = filePath
                message = message & vbLf & "State: " & eFile.CurrentState.Name
                message = message & vbLf & "Is Checked Out: " _
          & eFile.IsLocked.ToString
                If eFile.IsLocked Then
                    message = message & vbLf & "Is Checked Out By: " _
          & eFile.LockedByUser.Name
                End If

                'get the file's EnumeratorVariable interface
                'for working with its card variables
                Dim eVar As IEdmEnumeratorVariable10
                eVar = eFile.GetEnumeratorVariable

                'get the description
                Dim varValue As String
                eVar.GetVar("Description", "@", varValue)
                message = message & vbLf & "Description: " & varValue

                'try to check out the file
                Try
                    eFile.LockFile(parentFolder.ID, 0)
                Catch ex As Exception
                    MsgBox(ex.Message & vbLf & eFile.Name)
                    Exit Sub
                End Try

                'reconnect to the file's EnumeratorVariable
                'after check out
                eVar = eFile.GetEnumeratorVariable

                'edit the file here
                Try
                    eVar.SetVar("Description", "@", "NEW DESCRIPTION")
                    'close the file and save (flush) any updates
                    eVar.CloseFile(True)
                Catch ex As Exception
                    MsgBox(ex.Message & vbLf & eFile.Name)
                    Exit Sub
                End Try

                'try to check in the file
                Try
                    eFile.UnlockFile(0, "API check in")
                Catch ex As Exception
                    MsgBox(ex.Message & vbLf & eFile.Name)
                    Exit Sub
                End Try


                'show information about the selected file
                MsgBox(message)
            End While

        End If


    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
